src/preferences/preferences.cpp

```diff
@@ -309,24 +315,64 @@ void prefs::load_preferences()
 void prefs::write_preferences()
 {
 #ifndef _WIN32
-	bool prefs_file_existed = filesystem::file_exists(filesystem::get_prefs_file());
+	bool synced_prefs_file_existed = filesystem::file_exists(filesystem::get_prefs_file());
+	bool unsynced_prefs_file_existed = filesystem::file_exists(filesystem::get_unsynced_prefs_file());
 #endif
 
+	config synced;
+	config unsynced;
+
+	for(const char* attr : synced_attributes_) {
+		if(preferences_.has_attribute(attr)) {
+			synced[attr] = preferences_[attr];
+		}
+	}
+	for(const char* attr : synced_children_) {
+		for(const auto& child : preferences_.child_range(attr)) {
+			config& ch = synced.add_child(attr);
+			ch.append_children(child);
```

Comment:
Is this correct…? Something seems wrong about this. It adds a child tag to the synched prefs with the appropriate name, and then appends children from the child tag to that new tag… but how can that possibly be correct?

Reply:
When I didn't do that, it was dropping the `attr` child tag and its attributes (tested with the completed campaigns tags).

Reply:
More specifically, taking the example from the comments in `preferences.cpp`:
```
completed_campaigns = "A,B,C"
[completed_campaigns]
	[campaign]
		name = "A"
		difficulty_levels = "EASY,MEDIUM"
	[/campaign]
[/completed_campaigns]
```
it saves it as:
```
[completed_campaigns]
	[campaign]
		difficulty_levels="EASY,MEDIUM"
		name="A"
	[/campaign]
	[campaign]
		difficulty_levels=""
		name="B"
	[/campaign]
	[campaign]
		difficulty_levels=""
		name="C"
	[/campaign]
[/completed_campaigns]
```
and maintains it in that form no matter how many times I resave it. If I instead change it to `synced.append_children(child);` then it gets saved as:
```
[campaign]
	difficulty_levels="EASY,MEDIUM"
	name="A"
[/campaign]
[campaign]
	difficulty_levels=""
	name="B"
[/campaign]
[campaign]
	difficulty_levels=""
	name="C"
[/campaign]
```
after the first save and then removed entirely after the second save.

Reply:
Apparently this still doesn't preserve any attributes under the outermost tag either.

Reply:
What exactly is wrong with just `synced.add_child(attr, child)`?

Reply:
That works, apparently.